definitions/src/instructions.rs

```diff
@@ -86,158 +86,159 @@ macro_rules! __for_each_inst_inner {
-            (ORCB, 0x88),
-            (ORN, 0x89),
-            (REV8, 0x8a),
-            (ROL, 0x8b),
-            (ROLW, 0x8c),
-            (ROR, 0x8d),
-            (RORI, 0x8e),
-            (RORIW, 0x8f),
-            (RORW, 0x90),
-            (SEXTB, 0x91),
-            (SEXTH, 0x92),
-            (SH1ADD, 0x93),
-            (SH1ADDUW, 0x94),
-            (SH2ADD, 0x95),
-            (SH2ADDUW, 0x96),
-            (SH3ADD, 0x97),
-            (SH3ADDUW, 0x98),
-            (SLLIUW, 0x99),
-            (XNOR, 0x9a),
-            (ZEXTH, 0x9b),
+            (ADDUW, 0x63),
+            (ANDN, 0x64),
+            (BCLR, 0x65),
+            (BCLRI, 0x66),
+            (BEXT, 0x67),
+            (BEXTI, 0x68),
+            (BINV, 0x69),
+            (BINVI, 0x6a),
+            (BSET, 0x6b),
+            (BSETI, 0x6c),
+            (CLMUL, 0x6d),
+            (CLMULH, 0x6e),
+            (CLMULR, 0x6f),
+            (CLZ, 0x70),
+            (CLZW, 0x71),
+            (CPOP, 0x72),
+            (CPOPW, 0x73),
+            (CTZ, 0x74),
+            (CTZW, 0x75),
+            (MAX, 0x76),
+            (MAXU, 0x77),
+            (MIN, 0x78),
+            (MINU, 0x79),
+            (ORCB, 0x7a),
+            (ORN, 0x7b),
+            (REV8, 0x7c),
+            (ROL, 0x7d),
+            (ROLW, 0x7e),
+            (ROR, 0x7f),
+            (RORI, 0x80),
+            (RORIW, 0x81),
+            (RORW, 0x82),
+            (SEXTB, 0x83),
+            (SEXTH, 0x84),
+            (SH1ADD, 0x85),
+            (SH1ADDUW, 0x86),
+            (SH2ADD, 0x87),
+            (SH2ADDUW, 0x88),
+            (SH3ADD, 0x89),
+            (SH3ADDUW, 0x8a),
+            (SLLIUW, 0x8b),
+            (XNOR, 0x8c),
+            (ZEXTH, 0x8d),
             // Mop
-            (WIDE_MUL, 0x9c),
-            (WIDE_MULU, 0x9d),
-            (WIDE_MULSU, 0x9e),
-            (WIDE_DIV, 0x9f),
-            (WIDE_DIVU, 0xa0),
-            (FAR_JUMP_REL, 0xa1),
-            (FAR_JUMP_ABS, 0xa2),
-            (ADC, 0xa3),
-            (SBB, 0xa4),
-            (ADCS, 0xa5),
-            (SBBS, 0xa6),
-            (ADD3A, 0xa7),
-            (ADD3B, 0xa8),
-            (ADD3C, 0xa9),
-            (CUSTOM_LOAD_UIMM, 0xaa),
-            (CUSTOM_LOAD_IMM, 0xab),
+            (WIDE_MUL, 0x8e),
+            (WIDE_MULU, 0x8f),
+            (WIDE_MULSU, 0x90),
+            (WIDE_DIV, 0x91),
+            (WIDE_DIVU, 0x92),
+            (ADC, 0x93),
+            (SBB, 0x94),
+            (ADCS, 0x95),
+            (SBBS, 0x96),
+            (ADD3A, 0x97),
+            (ADD3B, 0x98),
+            (ADD3C, 0x99),
+            (CUSTOM_LOAD_UIMM, 0x9a),
+            (CUSTOM_LOAD_IMM, 0x9b),
+            // All branches
+            (AUIPC, 0x9c),
+            (BEQ, 0x9d),
+            (BGE, 0x9e),
+            (BGEU, 0x9f),
+            (BLT, 0xa0),
+            (BLTU, 0xa1),
+            (BNE, 0xa2),
+            (EBREAK, 0xa3),
+            (ECALL, 0xa4),
+            (FENCE, 0xa5),
```

Comment:
Why FENCE should be a  branch instruction?

Reply:
You are right they don't need to be branch instructions. The problem I want to solve, is that I believe it can benefit us from having some nop instructions that can manually **cut** instructions into traces. In other words, using `FENCE` here can manually mark the end of a trace, aiding better constructions of traces. This can unlock new potentials in the future IMHO. In reality, we won't see many occurences of fences as well, so I assume having them as branching instructions, won't affect current code.